objc/objc_runtime.go

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebiten Authors
+
+//go:build darwin
+// +build darwin
+
+// Package objc is a low-level pure Go objective-c runtime. This package is easy to use incorrectly so it is best
+// to use a wrapper that provides the functionality you need in a safer way.
+//
+// All functions that take a string as an argument are NULL terminated ('\x00'). This is so that there is no
```

Comment:
They are regular Go strings just need to add “\x00” to the end of it. And the package does verify that all the strings have been terminated properly. You can look at my Oto PR for what it looks like to actually use. I’m willing to allow non-null-terminated strings if you want but since calling into the runtime happens a LOT in Objc it would be best not to put extra pressure on the GC.  if you write the string as a constant with a null termination then the compiler is likely to put it in the data section and then there is no copying of data

Reply:
>  I’m willing to allow non-null-terminated strings if you want but since calling into the runtime happens a LOT in Objc it would be best not to put extra pressure on the GC. 

I cannot say anything without measuring, but I feel like this is a too early optimization. What about accepting regular strings and revisting this when we find this is really a problem?

Reply:
Well we can accept both! That way if someone finds that the GC pressure is too much then they can use null-terminated strings

Reply:
Ah yes, always checking the string be null-terminated or not and adding \x00 if needed would work :-) (Is my understanding correct?)

Reply:
Yep that is correct!